src/main/kotlin/no/nav/syfo/application/api/CronjobModule.kt

```diff
@@ -0,0 +1,45 @@
+package no.nav.syfo.application.api
+
+import io.ktor.application.*
+import kotlinx.coroutines.*
+import no.nav.syfo.application.ApplicationState
+import no.nav.syfo.application.Environment
+import no.nav.syfo.cronjob.journalforing.DialogmoteVarselJournalforingCronjob
+import no.nav.syfo.cronjob.leaderelection.LeaderPodClient
+
+fun Application.cronjobModule(
+    applicationState: ApplicationState,
+    environment: Environment
+) {
+    val leaderPodClient = LeaderPodClient(
+        environment = environment,
+    )
+    val journalforDialogmoteVarslerCronjob = DialogmoteVarselJournalforingCronjob(
+        applicationState = applicationState,
+        leaderPodClient = leaderPodClient,
+    )
+
+    createListenerCronjob(
+        applicationState = applicationState,
+    ) {
```

Comment:
Blir denne blokken action-parameteren i funksjonskallet?

Reply:
Ja

Reply:
Mer om High-order functions i kotlin for de som er interessert: https://kotlinlang.org/docs/lambdas.html

Reply:
Synes kanskje koden hadde vært letter å forstå dersom det hadde stått: 
```
    createListenerCronjob(
        applicationState = applicationState,
        {
            if (environment.journalforingCronjobEnabled) {
                journalforDialogmoteVarslerCronjob.start()
            } else {
               log.info("JournalforingCronjob is not enabled")
            }
        })
```
Eller er det noe jeg ikke skjønner?

Reply:
Er ikke det ca, det samme? Når jeg prøver det lokalt her nå, så gir IntelliJ og ber meg om å endre tilbake til det jeg hadde.

Reply:
Akkurat: "In Kotlin, there is a convention: if the last parameter of a function is a function, then a lambda expression passed as the corresponding argument can be placed outside the parentheses"